src/dss/logger.py

```diff
@@ -17,52 +17,51 @@ def format(self, record: logging.LogRecord) -> str:  # noqa: A003
 
 
 def setup_logger(
-    log_file_path: str, file_log_level: int = logging.DEBUG, console_log_level: int = logging.INFO
+    log_file_path: str = None,
+    file_log_level: int = logging.DEBUG,
+    console_log_level: int = logging.INFO,
 ) -> logging.Logger:
     """
-    Set up a logger with both file and console handlers.
+    Set up a logger with optional file and console handlers.
 
     Args:
-        log_file_path (str): Path to the log file.
-        file_log_level (int, optional): Logging level for file logs. Defaults to logging.DEBUG.
-        console_log_level (int, optional): Logging level for console logs. Defaults to logging.INFO.
+        log_file_path (str): Path to the log file. If None, uses $SNAP_COMMON/logs/dss.log.
+        file_log_level (int, optional): Logging level for file logs.
+        console_log_level (int, optional): Logging level for console logs.
 
     Returns:
-        logging.Logger: Configured logger object.
+        logging.Logger: Configured logger.
     """
     logger = logging.getLogger(__name__)
 
     if not logger.handlers:
-        logger.setLevel(file_log_level)
+        logger.setLevel(min(file_log_level, console_log_level))
 
-        # Formatter for console - omits [INFO]
+        # Console formatter
         console_formatter = CustomFormatter(
             "[%(levelname)s] %(message)s",
             datefmt="%Y-%m-%d %H:%M:%S",
         )
+        console_handler = logging.StreamHandler(sys.stdout)
+        console_handler.setLevel(console_log_level)
+        console_handler.setFormatter(console_formatter)
+        logger.addHandler(console_handler)
 
-        # Formatter for file - includes [INFO]
+        # Setup file handler if path is provided or fallback to SNAP_COMMON
+        if log_file_path is None:
+            snap_common = os.environ.get("SNAP_COMMON", "/tmp")
```

Comment:
Is there a chance that this targets `/tmp` but the [install hook](https://github.com/canonical/data-science-stack/pull/218/files#diff-e58b9a041906b37e03356984486699655b2d559fa8dc4d95b340a3b5be7a9a42R7) targets a different place (ie snap_commonis not set when the code is executed, but it is during the install hook)? Do you think this could pose a problem for the execution?

Reply:
This is not a problem as the python in dss runs as user so it should have access even without the hook.